### ISSUE TITLE
Code available?

### ISSUE THREAD

**Issue by ThomasLengeling** [hszhao/ICNet#1]
Hi Hengshuang,

I am wondering if by any chance your code is available for test?
Thank you

**Comment by hszhao** [comment]
Yeah, will be available no later than the end of May, possibly after nips ddl. Thanks.

**Comment by hszhao** [comment]
Yeah, will be available no later than the end of May, possibly after nips ddl. Thanks.

**Comment by weitaoatvison** [comment]
Hi, 
I am confused about the tricks  you made the ICNET to achieve at 30FPS on PSPNET50.  In your paper you introduced the details of compression method you choose. But In Section 6.1, you said "When directly compressing PSPNet50 following our previously described procedure, 170ms inference time is yielded with mIoU reducing to 67.9% as shown in Table 3. They indicate that only model compression has almost no chance to achieve realtime performance under the condition of keeping decent segmentation quality. In what follows, we take the model compressed PSPNet50, which is reasonably accelerated, as our baseline system for comparison." So, I am confused if  the speedup of the ICNET is come from the compressed PSPNet50 model you take  and not come from the compression method you take?

**Comment by StuvX** [comment]
My understanding is that the speedup is largely from an optimisation of model depth balanced against image resolution - so parallel inference is run using: 
1: full resolution through 4 layer model, 
2: 1/2 resolution through 8 layer model, and 
3: 1/4 resolution through 16 layer model. 
The inference is combined to produce the output in ~30 ms.

**Comment by ysh329** [comment]
mark

**Comment by johnsonice** [comment]
mark, looking forward for it.

**Comment by soeaver** [comment]
mark

**Comment by world4jason** [comment]
still waiting for u lol

**Comment by marshallixp** [comment]
mark

**Comment by tobschmidt** [comment]
any update?

**Comment by hszhao** [comment]
Dear all, thanks for the patience, the evaluation code is ready now.

**Comment by mehulmodihb** [comment]
HI, can you please let me know how to implement same feature on iOS/Android mobile application for background subtraction.
